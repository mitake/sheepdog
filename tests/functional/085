#!/bin/bash

# Test vdi repair functionality when inode object is broken

. ./common

for i in `seq 0 2`; do
    _start_sheep $i
done

_wait_for_sheep 3

$DOG node list

_cluster_format -i

$DOG vdi create test 12M
_vdi_list

echo "original data" | $DOG vdi write test

_random | dd of=$STORE/0/obj/807c2b2500000000 bs=4096 count=2048 &> /dev/null

$DOG cluster shutdown

for i in `seq 0 2`; do
    _start_sheep $i
done

_wait_for_sheep 3

$DOG vdi check test

$DOG vdi read test 0 14
