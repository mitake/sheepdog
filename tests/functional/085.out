QA output created by 085
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	128          0
   1   127.0.0.1:7001      	128          1
   2   127.0.0.1:7002      	128          2
using backend plain store
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
  test         0   12 MB  0.0 MB  0.0 MB DATE   7c2b25      3              
fixed replica 807c2b2500000000
finish check&repair test
original data
